src/components/Deployment/Deployment.tsx

```diff
@@ -0,0 +1,232 @@
+import { RoundedImage } from '../Image';
+import { useEtherspot } from '../../hooks';
+import Modal from '../Modal/Modal';
+import { Paragraph } from '../Text';
+import { PrimaryButton, SecondaryButton } from '../Button';
+import ErrorMessage from '../Error/ErrorMessage';
+
+//utils
+import { supportedChains } from '../../utils/chain';
+import { deployAccount } from '../../utils/transaction';
+import { Theme } from '../../utils/theme';
+
+const Wrapper = styled.div`
+  margin-top: 10px;
+`;
+
+const Section = styled.div`
+  display: flex;
+  background: ${({ theme }) => theme.color.background.topMenu};
+  margin-top: 5px;
+  padding: 15px;
+  border-radius: 10px;
+  font-weight: bold;
+`;
+
+const Header = styled.div`
+  display: block;
+`;
+
+const Body = styled.div`
+  display: block;
+`;
+
+const Label = styled.label`
+  display: inline-block;
+  margin-bottom: 14px;
+  font-size: 14px;
+`;
+
+const AuthLabel = styled.div`
+  background-color: ${({ theme }) => theme.color.background.button};
+  border-radius: 10px;
+  font-weight: bold;
+  padding: 5px;
+  position: absolute;
+  right: 28px;
+`;
+
+const DeployButton = styled.button`
+  background: ${({ theme }) => theme.color.background.deployButton};
+  color: ${({ theme }) => theme.color.text.listItemQuickButtonPrimary};
+  position: absolute;
+  font-weight: bold;
+  right: 28px;
+  border: none;
+  cursor: pointer;
+  border-radius: 10px;
+  height: 30px;
+  width: 80px;
+`;
+
+interface Account {
+  status: string;
+  value?: any;
+  title?: string;
+  address?: string;
+  createdAt?: Date;
+  state?: string;
+  store?: string;
+  type?: string;
+  updatedAt?: Date;
+}
+
+const Deployment = () => {
+  const [deployedChains, setDeployedChains] = useState<Account[]>([]);
+  const [undeployedChains, setUndeployedChains] = useState<Account[]>([]);
+  const [confirmModal, setConfirmModal] = useState<boolean>(false);
+  const [selectedChain, setSelectedChain] = useState<number | null>(null);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const { getSdkForChainId, accountAddress } = useEtherspot();
+  const theme: Theme = useTheme();
+
+  const fetchDeploymentData = async (chainId: number) => {
+    const sdk = getSdkForChainId(chainId);
+    if (accountAddress && sdk) {
+      try {
+        await sdk.computeContractAccount();
+        let account = await sdk.getAccount();
+        return account;
+      } catch (err) {
+        //
+      }
+    }
+  };
+
+  const getDeployementStatus = async () => {
+    try {
+      const result = await Promise.all(
+        supportedChains.map(async ({ chainId, title, iconUrl }) => {
+          const account = await fetchDeploymentData(chainId);
+          if (!account) return;
+          return { ...account, chainId, title, iconUrl };
+        })
+      );
+      const accounts = result.filter((acc) => acc);
+      return Promise.allSettled(accounts).catch(() => {
+        return [];
+      });
```

Comment:
accounts are not a promises, also what are you filtering here? accounts that are present?

```suggestion
      // filter out undefined
      return result.filter((acc) => !!acc);
```

Reply:
The purpose of filtering out accounts in this function is to remove any undefined or null values from the result array before passing it to Promise.allSettled() for further processing.